components/motor/gpio/motor_encoder.go

```diff
@@ -570,104 +422,108 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, revolutions float
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
 
+	m.state.regulated = true
 	if revolutions == 0 {
 		// Moving 0 revolutions is a special value meaning "move forever."
 		oldRpm := m.state.desiredRPM
 		m.state.desiredRPM = rpm
-		m.state.regulated = false // we're not going to a position
+		m.state.goalPos = math.Inf(int(rpm))
 
-		if math.Abs(oldRpm) > 0.001 && d == m.directionMovingInLock() {
+		// if we are already moving, let rpmMonitor deal with setPower
+		if math.Abs(oldRpm) > 0.001 && m.state.direction == m.directionMovingInLock() {
 			return nil
 		}
-		err := m.setPower(ctx, float64(d)*.06, true) // power of 6% is random
-		return err
+		// if moving from stop, start at 25% power
+		if err := m.setPower(ctx, m.state.direction*0.25, true); err != nil {
+			return err
+		}
+		return nil
 	}
 
-	numTicks := int64(revolutions * float64(m.ticksPerRotation))
-
-	pos, _, err := m.encoder.Position(ctx, encoder.PositionTypeUnspecified, nil)
+	currentPos, _, err := m.encoder.Position(ctx, encoder.PositionTypeUnspecified, nil)
 	if err != nil {
 		return err
 	}
-	m.state.setPoint = int64(pos) + d*numTicks
+	goalPos := (math.Abs(revolutions) * float64(m.ticksPerRotation) * m.state.direction) + currentPos
+
+	m.state.desiredRPM = math.Abs(rpm) * m.state.direction
+	m.state.goalPos = goalPos
 
-	_, rpmDebug := getRPMSleepDebug()
-	if rpmDebug {
-		m.logger.Debugf("received a goFor with rpm %0.1f, revolutions %0.1f and flip %d", rpm, revolutions, m.flip)
-		m.logger.Debugf("setpoint %d", m.state.setPoint)
+	startingPwr := 0.25 * m.state.direction
```

Comment:
So we just immediately set the power at the start of a gofor internal to a quarter %, what if we need to start slow? Happy to address this in another ticket.

Reply:
It was random before (0.06) to be fair.

Reply:
yeah this is reasonable for the rover but I see how that's probs not true for other motors, so I'll decrease it. But 6 is _too_ random I'm going to do 10 if that's alright

Reply:
We'll parametrize this based on the gofor call and the maxrpm using an S-curve in a separate ticket, and set it to a default otherwise